### PR TITLE
Arm reorg

### PR DESCRIPTION
**Pull request by We1chJ**

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -75,12 +75,76 @@ public final class Constants {
...
-    
+
     public static final int XBOX_BACK = 7;
     public static final int XBOX_START = 8;
 
     // joy stick button
     public static final int XBOX_JL = 9;
     public static final int XBOX_JR = 10;
 
+    // TODO: The following constants came from the 2022 robot.
```
Comment (prensing) [inline src/main/java/frc/robot/Constants.java]: Jack, please reference our discussion 1-2 weeks ago: The constants for a specific Subsystem (apart from CAN IDs) belong in the Subsystem class, not in Constants.

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -6,50 +6,43 @@
...
- * periodic methods (other than the scheduler calls). Instead, the structure of
- * the robot (including
- * subsystems, commands, and button mappings) should be declared here.
+ * This class is where the bulk of the robot should be declared. Since Command-based is a "declarative" paradigm, very
+ * little robot logic should actually be handled in the {@link Robot} periodic methods (other than the scheduler calls).
+ * Instead, the structure of the robot (including subsystems, commands, and button mappings) should be declared here.
  */
 public class RobotContainer {
-    // The robot's subsystems and commands are defined here...
+	// The robot's subsystems and commands are defined here...
```
Comment (prensing) [inline src/main/java/frc/robot/RobotContainer.java]: you have screwed up the indentation again. PLEASE look at the changes before you check them in.
Make sure to use the "Convert indentation to spaces" when you think it is messed up.

Comment (prensing) [reply src/main/java/frc/robot/RobotContainer.java]: Make sure you are not hiding whitespace changes.

File: src/main/java/frc/robot/commands/SetElevatorHeight.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import edu.wpi.first.wpilibj2.command.TrapezoidProfileCommand;
+import frc.robot.Constants;
+import frc.robot.subsystems.Arm;
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
+// information, see:
+// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html
+public class SetElevatorHeight extends CommandBase {
```
Comment (prensing) [inline src/main/java/frc/robot/commands/SetElevatorHeight.java]: The arm does not have a "height". Use "ArmLength".

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Arm extends SubsystemBase {
+	private Shoulder m_shoulder;
+	private Elevator m_elevator;
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/Arm.java]: I hate that we are calling it an Elevator. It does not go up and down. Need a better name.

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	public Arm() {
+		// Construct the shoulder and elevator trapezoid subsystems
+		m_shoulder = new Shoulder();
+		m_elevator = new Elevator();
+	}
+
+	public void periodic() {
+	}
+
+	public void setElevatorExtent(TrapezoidProfile.State extent) {
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/Arm.java]: What is "extent"?  "Length"??

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+		m_elevator.setSetPoint(extent);
+	}
+
+	// rotates the arms to a certain angle
+	public void setShoulderAngle(double degree) {
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/Arm.java]: "setAngle" is just as clear.

File: src/main/java/frc/robot/subsystems/Arm.java (hunk 5 above)
Comment (prensing) [reply src/main/java/frc/robot/subsystems/Arm.java]: "setLength"?

File: src/main/java/frc/robot/subsystems/Shoulder.java [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+        SmartDashboard.putNumber("Sim Battery voltage", RobotController.getBatteryVoltage());
+    }
+
+    @Override
+    public void periodic() {
+        double encoderValue = m_encoder.getIntegratedSensorAbsolutePosition();
+
+        // Display current values on the SmartDashboard
+        SmartDashboard.putNumber("shoulder/Output", m_motor.get());
+        SmartDashboard.putNumber("shoulder/Encoder", Units.radiansToDegrees(encoderValue));
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/Shoulder.java]: Math.toDegrees?

File: src/main/java/frc/robot/subsystems/Shoulder.java [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+    // = (2 * PI rads) / (4096 pulses)
+    public static final double kShoulderEncoderDistPerPulse = 2.0 * Math.PI / 4096;
+    // The arm gearbox represents a gearbox containing two Vex 775pro motors.
+    public final DCMotor m_shoulderGearbox = DCMotor.getFalcon500(2);
+
+    // Standard classes for controlling our arm
+    public final PIDController m_controller = new PIDController(m_kPShoulder, 0, 0);
+
+    // Simulation classes help us simulate what's going on, including gravity.
+    public static final double m_shoulderReduction = 395.77;
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/Shoulder.java]: Why are any of these public?

File: src/main/java/frc/robot/subsystems/Shoulder.java [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+        double p = SmartDashboard.getNumber("shoulder/P Gain", 0);
+        // if PID coefficients on SmartDashboard have changed, write new values to
+        // controller
+        if ((p != m_kPShoulder)) {
+            m_motor.config_kP(Constants.kPIDLoopIdx, Constants.SHOULDER_K_P, Constants.kTimeoutMs);
+            m_kPShoulder = p;
+        }
+    }
+
+    public WPI_TalonFX getMotor() {
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/Shoulder.java]: Why does this exist? No one outside this class should be accessing the motor.

File: src/main/java/frc/robot/subsystems/Shoulder.java [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+        return m_motor;
+    }
+
+    public TalonFXSensorCollection getEncoder() {
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/Shoulder.java]: same. This should not exist.

File: .vscode/settings.json [hunk 11]
```diff
@@ -26,7 +26,7 @@
     },
   ],
   "java.test.defaultConfig": "WPIlibUnitTests",
-  "editor.tabSize": 8,
+  "editor.tabSize": 4,
```
Comment (prensing) [inline .vscode/settings.json]: Don't change the value of a tab. A bad practice and just masks the incorrect indentation.

**Review by prensing — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (We1chJ) [reply src/main/java/frc/robot/Constants.java]: I will merge out from main once your new branch that reorganizes constants is merged to main

Comment (We1chJ) [reply src/main/java/frc/robot/Constants.java]: Nevermind, I will move it

File: src/main/java/frc/robot/RobotContainer.java [hunk 12]
```diff
@@ -48,15 +43,16 @@ public class RobotContainer {
     private final XboxController m_controller = new XboxController(0);
 
     // The robot's subsystems and commands are defined here...
-    private final Vision m_vision = new Vision();
-    private final DriveTrain m_driveTrain = new DriveTrain(m_vision);
+    // private final Vision m_vision = new Vision();
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/RobotContainer.java]: If this is suppoed to work on the real robot, we 'll want to keep this. Either put it back or, if we're just testing, add a TODO comment to remind us.

File: src/main/java/frc/robot/RobotContainer.java [hunk 13]
```diff
@@ -48,15 +43,16 @@ public class RobotContainer {
...
+    // private final DriveTrain m_driveTrain = new DriveTrain(m_vision);
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/RobotContainer.java]: If this is suppoed to work on the real robot, we 'll want to keep this. Either put it back or, if we're just testing, add a TODO comment to remind us.

File: src/main/java/frc/robot/RobotContainer.java [hunk 14]
```diff
@@ -48,15 +43,16 @@ public class RobotContainer {
...
+	private final Arm m_arm = new Arm();
 
     /**
      * The container for the robot. Contains subsystems, OI devices, and commands.
      */
     public RobotContainer() {
         configureButtonBindings();
-        m_driveTrain.setDefaultCommand(getDriveCommand());
+        // m_driveTrain.setDefaultCommand(getDriveCommand());
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/RobotContainer.java]: If this is suppoed to work on the real robot, we 'll want to keep this. Either put it back or, if we're just testing, add a TODO comment to remind us.

**Review by prensing — APPROVED** [review]

**Pull request merged**